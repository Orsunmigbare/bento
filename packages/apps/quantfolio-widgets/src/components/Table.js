import React, { useState, useEffect } from "react"
import styled from "styled-components"
import { theming, Box, Heading } from "@staccx/base"

const Table = ({ data }) => {
  return (
    <Box>
      <Heading>Macro Table</Heading>
      <StyledTable>
        <thead>
          <tr>
            <td>Indicator</td>
            <td>Status</td>
            <td>Updated</td>
            <td>Current</td>
            <td>Treshhold</td>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td>a</td>
            <td>b</td>
            <td>c</td>
            <td>d</td>
            <td>d</td>
          </tr>
          <tr>
            <td colSpan="5">a</td>
          </tr>
        </tbody>
        <tbody>
          <tr>
            <td>a</td>
            <td>b</td>
            <td>c</td>
            <td>d</td>
            <td>d</td>
          </tr>
          <tr>
            <td colSpan="5">a</td>
          </tr>
        </tbody>
      </StyledTable>
    </Box>
  )
}

const StyledTable = styled.table`
  width: 100%;

  > thead {
    font-weight: bold;
  }

  > tbody {
    &:nth-of-type(even) {
      background-color: ${theming.color.gray};
    }
  }
`

export default Table
