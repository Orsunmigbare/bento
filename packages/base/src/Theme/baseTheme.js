import { ThemeManager } from "@staccx/theme"

export const theme = {
  name: "Default theme",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px",
    micro: "3px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px",
    breakout: "640px"
  },
  color: {
    white: "#fff",
    bg: "#fff",
    subtleHover: "#F7F7FF",
    bgGray: "#f7f7ff",
    line: "#EBEBF2",
    gray: "#767676",
    wcag: "#777",
    disabled: "#BABABA",
    primary: "#00a190",
    secondary: "#F5A623",
    text: "#2b2b2b",
    negative: "#ea526f",
    positive: "#4A90E2",
    warning: "#c54059",
    black: "#000",
    green: "#00a190" // for toggle eks
  },
  fontFamily: {
    heading: "sans-serif",
    body: "sans-serif"
  },
  font: {
    h1: "36px",
    h2: "26px",
    h3: "22px",
    h4: "18px",
    h5: "16px",
    h6: "14px",
    base: "14px",
    tiny: "12px",
    input: "16px",
    huge: "44px"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

const manager = new ThemeManager(theme)
export default manager
