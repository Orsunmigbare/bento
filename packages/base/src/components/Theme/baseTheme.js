export const theme = {
  name: "StaccXBase",
  spacing: {
    huge: "96px",
    large: "48px",
    mediumPlus: "32px",
    medium: "24px",
    small: "12px",
    tiny: "6px"
  },
  wrapper: {
    small: "420px",
    medium: "640px",
    large: "1024px"
  },
  color: {
    bg: "#fff",
    bgGray: "#f7f7ff",
    primary: "#00a190",
    gray: "#767676",
    text: "#2b2b2b",
    white: "#fff",
    red: "#ea526f",
    green: "#00a190",
    blue: "#988EFB",
    yellow: "#f4da87",
    line: "#EBEBF2",
    wcag: "#777",
    disabled: "#BABABA",
    warning: "#c54059",
    black: "#000",
    subtleHover: "#F7F7FF"
  },
  font: {
    size: {
      h1: "36px",
      h2: "26px",
      h3: "22px",
      h4: "18px",
      h5: "16px",
      h6: "14px",
      base: "14px",
      tiny: "12px",
      input: "16px",
      huge: "44px"
    },
    heading: "Libre Baskerville",
    body: "Libre Franklin"
  },
  targetSize: {
    small: "36px",
    normal: "44px"
  },
  borderRadius: "2px"
}

export default theme
