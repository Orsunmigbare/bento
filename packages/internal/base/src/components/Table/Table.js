import React from "react"
import PropTypes from "prop-types"
import {
  // color,
  // borderRadius,
  // font,
  // fontFamily,
  // fontWeight,
  // spacing,
  // targetSize,
  themify
} from "@staccx/theme"
import styled from "styled-components"
import themePropTypes from "../constants/themePropTypes"

class Table extends React.Component {
  render() {
    const {
      data,
      blacklist,
      mapHeader,
      children,
      variant,
      itemToKey,
      onHeaderClick,
      renderHeader,
      ...restProps
    } = this.props

    const headers = data
      .reduce((acc, current) => {
        Object.keys(current).forEach(key => {
          if (!acc.includes(key)) {
            acc.push(key)
          }
        })

        return acc
      }, [])
      .filter(blacklist)
      .map(header => ({
        value: header,
        title: header
      }))
      .map(mapHeader)

    return (
      <TableStyled variant={variant} {...restProps}>
        <thead>
          <tr>
            {headers.map((item, index) => {
              if (renderHeader) {
                return renderHeader({ item, index })
              }
              return (
                <th
                  key={`thead-${item.value}`}
                  onClick={e => onHeaderClick(e, item)}
                >
                  {item.title}
                </th>
              )
            })}
          </tr>
        </thead>
        <tbody>
          {this.props.data.map((item, index) => {
            const key = itemToKey(item)
            return (
              <tr key={`tr-${typeof key === "string" ? key : index}`}>
                {children({ item, headers, index })}
              </tr>
            )
          })}
        </tbody>
      </TableStyled>
    )
  }
}

Table.themeProps = {
  table: {
    name: "base-component-table",
    description: "",
    type: themePropTypes.style
  }
}

export const TableStyled = styled.table`
  ${themify(Table.themeProps.table)};
`

Table.propTypes = {
  blacklist: PropTypes.func,
  children: PropTypes.func.isRequired,
  data: PropTypes.array.isRequired,
  fallbackCellValue: PropTypes.any,
  itemToKey: PropTypes.any,
  mapHeader: PropTypes.func,
  onHeaderClick: PropTypes.func,
  renderHeader: PropTypes.func
}

export default Table

Table.defaultProps = {
  blacklist: item => item,
  data: [{ name: "one" }, { name: "two", lol: "omg" }],
  fallbackCellValue: "n/a",
  itemToKey: item => item,
  mapHeader: item => item,
  onHeaderClick: console.log
}
