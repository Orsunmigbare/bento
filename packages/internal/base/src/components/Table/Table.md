### Vanilla use
```jsx harmony
<Table>
{({value}) => (<td><h3>{value}</h3></td>)}
</Table>



```

### Render header
```js
<Table renderHeader={({item}) => (<th><i>{item.value}</i></th>)}>
{({value}) => (<td><h3>{value}</h3></td>)}
</Table>
```