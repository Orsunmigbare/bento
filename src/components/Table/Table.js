import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"
import { applyVariants, themePropTypes } from "../../theming"

class Table extends React.Component {
  render() {
    const {
      data,
      blacklist,
      mapHeader,
      children,
      variant,
      itemToKey,
      onHeaderClick,
      renderHeader,
      overrideHeaders,
      ...restProps
    } = this.props

    const headers = overrideHeaders
      ? overrideHeaders
          .map(header => ({
            value: header,
            title: header
          }))
          .map(mapHeader)
      : data
          .reduce((acc, current) => {
            Object.keys(current).forEach(key => {
              if (!acc.includes(key)) {
                acc.push(key)
              }
            })

            return acc
          }, [])
          .filter(blacklist)
          .map(header => ({
            value: header,
            title: header
          }))
          .map(mapHeader)

    return (
      <TableStyled variant={variant} {...restProps}>
        <thead>
          <tr>
            {headers.map((item, index) => {
              if (renderHeader) {
                return renderHeader({ item, index })
              }
              return (
                <th
                  key={`thead-${item.value}`}
                  onClick={e => onHeaderClick(e, item)}
                >
                  {item.title}
                </th>
              )
            })}
          </tr>
        </thead>
        <tbody>
          {this.props.data.map((item, index) => {
            const key = itemToKey(item)
            return (
              <tr key={`tr-${typeof key === "string" ? key : index}`}>
                {children({ item, headers, index })}
              </tr>
            )
          })}
        </tbody>
      </TableStyled>
    )
  }
}

Table.themeProps = {
  table: {
    name: "base-component-table",
    description: "",
    type: themePropTypes.style
  }
}

export const TableStyled = styled.table`
  ${applyVariants(Table.themeProps.table)};
`

Table.propTypes = {
  blacklist: PropTypes.func,
  children: PropTypes.func.isRequired,
  data: PropTypes.array.isRequired,
  fallbackCellValue: PropTypes.any,
  itemToKey: PropTypes.any,
  mapHeader: PropTypes.func,
  onHeaderClick: PropTypes.func,
  renderHeader: PropTypes.func,
  overrideHeaders: PropTypes.array
}

export default Table

Table.defaultProps = {
  blacklist: item => item,
  data: [{ name: "one" }, { name: "two", lol: "omg" }],
  fallbackCellValue: "n/a",
  itemToKey: item => item,
  mapHeader: item => item,
  onHeaderClick: console.log
}
