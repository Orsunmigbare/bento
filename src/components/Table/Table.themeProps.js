import { themePropTypes } from "../../theming"

const themeProps = {
  table: {
    name: "base-component-table",
    description: "",
    type: themePropTypes.style
  }
}

export default themeProps
