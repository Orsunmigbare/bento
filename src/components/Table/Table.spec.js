import React from "react"
import renderer from "react-test-renderer"
import { ThemeProvider } from "styled-components"
import baseTheme from "../../theming/themes/baseTheme"
import Table from "./Table"

describe("Table", () => {
  describe("Snapshots", () => {
    it("Table", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Table
              renderHeader={({ item }) => (
                <th>
                  <i>{item.value}</i>
                </th>
              )}
            >
              {({ value }) => (
                <td>
                  <h3>{value}</h3>
                </td>
              )}
            </Table>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
    it("Tab", () => {
      const tree = renderer
        .create(
          <ThemeProvider theme={baseTheme}>
            <Table onClick={e => console.log(e.target.value)}>
              {({ value }) => (
                <td onClick={e => console.log(e.target.value)}>
                  <h3>{value}</h3>
                </td>
              )}
            </Table>
          </ThemeProvider>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
