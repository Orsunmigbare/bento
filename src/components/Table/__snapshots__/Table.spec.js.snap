// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Snapshots Tab 1`] = `
<table
  className="sc-AykKC QXPLA"
  onClick={[Function]}
>
  <thead>
    <tr>
      <th
        onClick={[Function]}
      >
        name
      </th>
      <th
        onClick={[Function]}
      >
        lol
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        onClick={[Function]}
      >
        <h3 />
      </td>
    </tr>
    <tr>
      <td
        onClick={[Function]}
      >
        <h3 />
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Table Snapshots Table 1`] = `
<table
  className="sc-AykKC QXPLA"
>
  <thead>
    <tr>
      <th>
        <i>
          name
        </i>
      </th>
      <th>
        <i>
          lol
        </i>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        <h3 />
      </td>
    </tr>
    <tr>
      <td>
        <h3 />
      </td>
    </tr>
  </tbody>
</table>
`;
